plugins/modules/aci_vmm_controller.py

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Manuel Widmer <mawidmer@cisco.com>
+# Copyright: (c) 2021, Anvitha Jain (@anvitha-jain) <anvjain@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_vmm_controller
+short_description: Manage Controller to domain association for virtual domains profiles (vmm:DomP)
+description:
+- Manage vCenter virtual domains on Cisco ACI fabrics.
+  Currently only VMware vSphere Distributed Switch is supported via this module.
+options:
+  name:
+    description:
+    - Name of VMM Controller.
+    type: str
+    aliases: []
+  controller_hostname:
+    description:
+    - Hostname of IP of the controller.
+    type: str
+    aliases: []
+  dvs_version:
+    description:
+    - Version of the VMware DVS.
+    type: str
+    aliases: []
+    choices: [ 'unmanaged', '5.1', '5.5', '6.0', '6.5', '6.6', '7.0' ]
+  stats_collection:
+    description:
+    - Whether stats collection is enabled.
+    type: str
+    choices: [ 'enabled', 'disabled' ]
+    default: disabled
+    aliases: []
+  domain:
+    description:
+    - Name of the virtual domain profile.
+    type: str
+    aliases: [ domain_name, domain_profile ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+  credentials:
+    description:
+    - Name of the VMM credentials to be used
+    type: str
+  management_epg:
+    description:
+    - Name of the management EPG to be used by the controller. Only supports in-band management EPGs for now.
```

Comment:
How much work would it be to add support for out of band?

Reply:
no out of bound support available yet